### PR TITLE
api(dotnet): introduce RunAndWaitForAsync

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/api/class-browsercontext.md [hunk 1]
```diff
@@ -1224,9 +1225,10 @@ page = event_info.value
 ```
 
 ```csharp
-var waitForPageEvent = context.WaitForPageAsync();
-await page.ClickAsync("button");
-var page = await waitForPageEvent;
+var page = await context.RunAndWaitForEventAsync(ContextEvent.Page, async _ =>
```
Comment (dgozman) [inline docs/src/api/class-browsercontext.md:1228]: nit:  `async ()` instead of `async _` ?

File: docs/src/api/class-browsercontext.md [hunk 2]
```diff
@@ -1259,3 +1261,21 @@ Will throw an error if the context closes before new [Page] is created.
...
+  - alias-csharp: WaitForEventAsync
+- returns: <[any]>
+
+:::note
+In most cases, you should use [`method: BrowserContext.waitForEvent`].
+:::
+
+Waits for given `event` to fire. If predicate is provided, it passes
+event's value into the `predicate` function and waits for `predicate(event)` to return a truthy value.
+Will throw an error if the socket is closed before the `event` is fired.
```
Comment (dgozman) [inline docs/src/api/class-browsercontext.md:1277]: socket is closed -> browser context is closed

File: docs/src/api/class-page.md [hunk 3]
```diff
@@ -3694,3 +3695,21 @@ associated with the page.
...
+  - alias-csharp: WaitForEventAsync
+- returns: <[any]>
+
+:::note
+In most cases, you should use [`method: Page.waitForEvent`].
+:::
+
+Waits for given `event` to fire. If predicate is provided, it passes
+event's value into the `predicate` function and waits for `predicate(event)` to return a truthy value.
+Will throw an error if the socket is closed before the `event` is fired.
```
Comment (dgozman) [inline docs/src/api/class-page.md:3711]: socket is closed -> page is closed

File: docs/src/api/params.md [hunk 4]
```diff
@@ -480,10 +480,17 @@ is considered matching if all specified properties match.
...
+* langs: js, java, python
 - `event` <[string]>
 
 Event name, same one typically passed into `*.on(event)`.
 
+## csharp-wait-for-event-event
+* langs: csharp
+- `playwrightEvent` <[PlaywrightEvent<T>]>
+
+Event type, same one typically passed into `WaitForEventAsync`.
```
Comment (dgozman) [inline docs/src/api/params.md:492]: This is the main method we'll show in the snippets? I think it should not redirect to some other method for the parameter description.

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-browsercontext.md (hunk 1 above)
Comment (kblok) [reply docs/src/api/class-browsercontext.md]: He needs to discard the event argument

**Pull request merged**